### ISSUE TITLE
Support for Play 2.6

### ISSUE THREAD

**Issue by JakubKahovec** [t2v/play2-auth#184]
Hello,

Do you plan to add support for Play 2.6.x

Thank you

**Comment by Daxten** [comment]
@t2v 2.6 is out now, will you update this library or will someone has to fork?

best regards

**Comment by djx314** [comment]
+1

**Comment by ezhulkov** [comment]
It would be highly appreciated
+1

**Comment by dancetrain** [comment]
Looking forward to see 2.6 play framework support

**Comment by gakuzzzz** [comment]
Thanks for open this issue.

And I'm sorry about the delay in integration to Play 2.5 over.

We are planning for create play2-auth version2 for Play2.6. 

It will be DI based composition instead of mix-in based composition.

Probably it can not keep the source compatibility to play2-auth 0.x.

However I think that it is better way to integrate to current Play2 ecosystem.

I would like to hear your opinions about it.

best regards

**Comment by Daxten** [comment]
exactly what I thought,
RequestWithAttributes shouldn't be necessary anymore for example (see: https://www.playframework.com/documentation/2.6.x/Highlights26#Request-attributes)

**Comment by gakuzzzz** [comment]
Yes, the next play2-auth should use the Request attribute API built-in Play2

**Comment by ezhulkov** [comment]
@gakuzzzz nice news, thank you
Just wondering, how are you going to compose injected actions? Inject many ActionBuilders into Controller and then use something like 
`builder1 andThen builder2` 
to get resulting Action?

**Comment by Daxten** [comment]
Hi,
https://github.com/Daxten/bay-scalajs.g8/blob/e447955c34123e04b0298db931b3e88a14abaa4b/server/app/controllers/Security.scala

this is me just trying out the actionBuilder stuff, but this can create nearly the same syntax then before

I would keep all the builders private or in a nested class and compose them before handing out the dsl to the user

**Comment by gakuzzzz** [comment]
@ezhulkov @Daxten
This is an annoying problem.
ActionFunctions what have a same Request type can compose each other.
However Different Request type ActionFunctions can not compose.
And ActionBuilders can not compose other ActionBuilder.

To tell the truth, play2-auth has ActionFunctions what partially solves this problem
https://github.com/t2v/play2-auth/blob/master/README.ja.md#actionfunction-%E3%81%A8%E3%81%97%E3%81%A6%E3%81%AE%E5%88%A9%E7%94%A8 (Japanese)

Currently, I think that a better way to compose ActionBuilders is like the [action-zipper](https://github.com/t2v/action-zipper) approach.

**Comment by Daxten** [comment]
I don't think that should be a problem, you can use a sealed trait for example and create a more generic ActionFunction that way

**Comment by gakuzzzz** [comment]
@Daxten
play2-auth's ActionBuilder should be enable to compose all ActionBuilders (provided by any libraries which play2-auth does not know)

**Comment by Daxten** [comment]
Ah! Now I see the problem, yes that's shitty design then by play :/

**Comment by ezhulkov** [comment]
Agree, action composition in play looks not very convinient
 
> And ActionBuilders can not compose other ActionBuilder.
You can try to compose many ActionBuilders using andThen or compose syntax. Smth like this https://gist.github.com/ezhulkov/90d050a0701206d318999f2995f14472#file-testcontroller-scala-L8

May be it will be better and more concise to have current approach with super() calls and trait chains?

I've forked recently you stackable-controller and put some fixes. 
Have not got a chance to get rid of RequestWithAttributes, but it's coming.

https://github.com/zeptolab/stackable-controller/blob/play26/core/src/main/scala/com/zeptolab/play2/stackc/StackableController.scala

**Comment by Daxten** [comment]
@gakuzzzz how are you doing? :)

**Comment by ezhulkov** [comment]
Any news? Scala 2.12.3 and play 2.6.3 are waiting for t2v support :)

**Comment by fdoumet** [comment]
We built on Scala 2.12 and Play 2.6 using [this](https://github.com/gitsumanth/play2-auth/tree/feature/play2.6) branch.

**Comment by Daxten** [comment]
would you mind to release it somewhere (bintray is incredibly easy for example) ?

**Comment by nafg** [comment]
@fdoumet can you open a pull request at least?

**Comment by ptrdom** [comment]
@fdoumet some sort of release would be amazing!

**Comment by lu4nm3** [comment]
What's the timeline looking like for releasing support for Play 2.6?

**Comment by andrewswan** [comment]
It's a few months later - any news please?

**Comment by Enalmada** [comment]
Looks like there is a pull request: https://github.com/t2v/play2-auth/pull/191


I recommend you migrate to silhouette which I now use in all new projects.  It's much more complicated but also well maintained.    I would ask the author of play2-auth to just merge the pull request and be done with it.  Don't worry about maintaining pre-2.6 branches as the current release version is fine for pre-2.6 and there are critical fixes in 2.6 that everyone should be up to by now anyway.

**Comment by fdoumet** [comment]
If anybody's still looking for this, here are the details:
https://bintray.com/adenda/play2-auth/play2-auth
